src/main/java/org/wildfly/security/auth/realm/FileSystemSecurityRealm.java

```diff
@@ -574,6 +592,12 @@ private void writeIdentity(final XMLStreamWriter streamWriter, final LoadedIdent
             streamWriter.writeEndDocument();
         }
 
+        public void dispose() {
+            // Release the write lock for this realm identity
+            final ReentrantReadWriteLock realmIdentityLock = getRealmIdentityLockForName(name);
+            realmIdentityLock.writeLock().unlock();
```

Comment:
I would suggest java.nio.channels.FileLock but unfortunately, while they do control locking between processes, they cannot be used to block other threads in the same process that created the lock.